### ISSUE TITLE
Auto Compaction event itself triggers compaction error log? (3.4.9)

### ISSUE THREAD

**Issue by paulcaskey** [etcd-io/etcd#12175]
I know "[Revisions prior to the compaction revision become inaccessible](https://etcd.io/docs/v3.4.0/op-guide/maintenance/)" so a request for an old key history can trigger this error. **required revision has been compacted**
But - sometimes - every host in my cluster now throws this, every hour at the exact same second. v3.4.9.

The compaction event seems to be complaining about itself? There is no reference to a watch or get event in the log, although I suspect this happens when I have an active watch going. I'm not actually sure what triggers this. It doesn't happen all the time.

```
ETCD_AUTO_COMPACTION_MODE=periodic
ETCD_AUTO_COMPACTION_RETENTION=60h
```
`Jul 26 13:54:56 host.foo.com etcd[42348]: failed to apply request "header:<ID:3529259843468490369 > compaction:<revision:1574763 > " with response "" took (18.467µs) to execute, err is mvcc: required revision has been compacted`

**Comment by paulcaskey** [comment]
I tried updating to v3.4.10 and see no difference fwiw. Here is a brand new install with no data and no client traffic. Simple 3-node cluster. Still I see these regular warnings. Here I have ETCD_AUTO_COMPACTION_RETENTION=5m so I see this `err is mvcc: required revision has been compacted` every 300 seconds on all 3 nodes. 

I can't figure out what I'm doing something strange or unusual, or else everyone would be noticing this. Help?
```
Aug 10 09:20:43 host.foo.com etcd[171495]: Finished auto-compaction at revision 1
Aug 10 09:25:43 host.foo.com etcd[171495]: Starting auto-compaction at revision 1 (retention: 5m0s)
Aug 10 09:25:43 host.foo.com etcd[171495]: failed to apply request "header:<ID:11862608794933001224 > compaction:<revision:1 > " with response "" took (11.086µs) to execute, err is mvcc: required revision has been compacted
Aug 10 09:25:43 host.foo.com etcd[171495]: Finished auto-compaction at revision 1
Aug 10 09:30:43 host.foo.com etcd[171495]: Starting auto-compaction at revision 1 (retention: 5m0s)
Aug 10 09:30:43 host.foo.com etcd[171495]: failed to apply request "header:<ID:11862608794933001225 > compaction:<revision:1 > " with response "" took (10.192µs) to execute, err is mvcc: required revision has been compacted
Aug 10 09:30:43 host.foo.com etcd[171495]: Finished auto-compaction at revision 1
Aug 10 09:35:43 host.foo.com etcd[171495]: Starting auto-compaction at revision 1 (retention: 5m0s)
Aug 10 09:35:43 host.foo.com etcd[171495]: failed to apply request "header:<ID:11862608794933001226 > compaction:<revision:1 > " with response "" took (9.846µs) to execute, err is mvcc: required revision has been compacted
Aug 10 09:35:43 host.foo.com etcd[171495]: Finished auto-compaction at revision 1
Aug 10 09:40:43 host.foo.com etcd[171495]: Starting auto-compaction at revision 1 (retention: 5m0s)
Aug 10 09:40:43 host.foo.com etcd[171495]: failed to apply request "header:<ID:11862608794933001227 > compaction:<revision:1 > " with response "" took (11.533µs) to execute, err is mvcc: required revision has been compacted

```

**Comment by paulcaskey** [comment]
Theory: This only happens when there are no PUTs between auto compaction runs. It's an off-by-one error on a counter that didn't increment at all. Will verify and fix if I can.

**Comment by tangcong** [comment]
I added a warning log when failed to apply request in v3.4.9,It helped us troubleshooting serious bugs. The etcd revision will not change when there is no write request, but periodic compactor will be called periodic, so this warning log will appear(rev == s.compactMainRev). I think this error can be ignored. For the compact RPC Method, returning nil and ErrCompacted are both successful . i submitted pr #12211. thanks. 

```
	if rev <= s.compactMainRev {
		ch := make(chan struct{})
		f := func(ctx context.Context) { s.compactBarrier(ctx, ch) }
		s.fifoSched.Schedule(f)
		s.revMu.Unlock()
		return ch, ErrCompacted
	}
```

**Comment by jingyih** [comment]
Is this already fixed by #12211?
